p2p.py

```diff
@@ -102,15 +102,16 @@ def print_investment_data(investment_data):
             logger.info(f'  {date}: {file_path}')
 
 
-def filter_investment_files_by_date(investment_data, dates:list = []):
+def filter_investment_files_by_date(investment_data, kind:str = 'list', dates:list = []):
```

Comment:
This function should be redone/splitted, I just missed it in the previous review. It is not following the single responsibility principal. The best way to see it is that you need to give an additional parameter to say what the function should do.

Reply:
About setting the list as a default as a parameter is not a good practice for several reasons:
https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html